### DISCUSSION TITLE
Date determination folder upload

### DISCUSSION THREAD

**Discussion by Sithuk** [simulot/immich-go#483] [General]
The readme notes that a photo's date is determined by date information in: XMP file, filename, path, exif data. In that order. 

I have collated photos in folders that have the date when the folder was created, despite when the photos may have been taken. The photo filenames have no date/time information. 

Does that mean if I upload the photos to immich with immich-go that the photos will the folder date as their date of capture (there are no XMP files)? Is there anyway to use the date in the photo exif instead?

**Comment by simulot** [comment]
The readme isn't precise enough:

I have looked at the code, and the XMP is not used by immich-go. However when existing, the XMP file is passed to immich, and immich uses it as primary source of metadata. Therefore if the XMP contains the date of capture it will be used.

If there isn't any XMP files, the date is determined by the file name if possible the the path, then the exif.

The next major release will give a better control on the date determination.

**Reply by Sithuk** [reply]
Thank you, so if there is a date in the path, that date will override the date in the photo exif information?

**Reply by Sithuk** [reply]
I'm curious why a date needs to be provided to Immich. 

I appreciate it may be useful for scenarios where the date information is missing from the exif info or from an XMP file. But otherwise, Immich is able to extract the date itself when I import a photo. So why does Immich-Go need to provide a date?

Thank you for the update on the next major release and that it will have an option to identify a priority for the date determination. Do you have an expected timeframe for release of the next version?

**Reply by Sithuk** [reply]
I have tested the import of a photo with a different date to the date of the folder that it is in. I can confirm that Immich shows the folder date, not the date in the EXIF information. So the date in the EXIF information is ignored / not used. I look forward to the next version of Immich-go where hopefully we will be able to prioritise the EXIF information over any date information in the path / folder name.

**Reply by simulot** [reply]
>  I'm curious why a date needs to be provided to Immich.

It's a free way to give a capture date to old scanned images, and old photos patiently placed in folders named after the date. This fits well with my personal use case.

Now, as the project gains popularity, users may want different options.  This is why the next major release introduce an option to control how the date is captured by immich-go: NONE, EXIF,  FILENAME, EXIF-FILENAME and FILENAME-EXIF.

**Reply by Sithuk** [reply]
Giving a date to old photos with no exif information makes sense. Might be better to prioritise the use of exif and xmp over the path information? 

Some photos could include exif date information only and some might have no exif metadata and have the date incorporated in the folder name. A flag to force a single date source for an entire bulk upload is likely to be suboptimal. Better to use the exif or xmp info if present, then fallback to the path info if not?

**Reply by emulated24** [reply]
> > I'm curious why a date needs to be provided to Immich.
> 
> It's a free way to give a capture date to old scanned images, and old photos patiently placed in folders named after the date. This fits well with my personal use case.
> 
> Now, as the project gains popularity, users may want different options. This is why the next major release introduce an option to control how the date is captured by immich-go: NONE, EXIF, FILENAME, EXIF-FILENAME and FILENAME-EXIF.

Now this would be awesome, I'm just importing my library to fresh Immich server and struggle with photo timestamp. I would prefer EXIF over anything else as I can manipulate that to whatever I prefer.
Any ETA on the functionality implementation please?

**Reply by simulot** [reply]
I'm working on it, best effort mode. No ETA sorry
